### DISCUSSION TITLE
tracing-appender mangles text

### DISCUSSION THREAD

**Discussion by SephVelut** [tokio-rs/tracing#2589] [Q&A] [answered]
```
fn main() {
    let file_appender = tracing_appender::rolling::never("logs", "log");
    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
    tracing_subscriber::fmt()
        .with_writer(non_blocking)
        .init();

    error!("write something");
}
```

./logs/log file has this text
`[2m2023-05-10T16:43:35.091245Z[0m [31mERROR[0m [2msip[0m[2m:[0m write something`

**Comment by hawkw** [comment, marked as answer]
The text in your log file appears mangled because it contains [ANSI escape codes](https://en.m.wikipedia.org/wiki/ANSI_escape_code). When logging to the terminal, these escape codes configure formatting, such as changing the color of the text and italicizing, bolding, or underlining text. When written to a text file, however, these escape codes do nothing. 

By default, `tracing_subscriber::fmt` assumes that logs will be written directly to the terminal, but you are instead configuring it to write to a log file. You can call [`with_ansi(false)`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.SubscriberBuilder.html#method.with_ansi) when constructing the subscriber in order to disable the use of these escape codes.

Change the subscriber construction to something like this, and your log file should look much nicer:
```rust
    tracing_subscriber::fmt()
        .with_writer(non_blocking)
        .with_ansi(false)
        .init();
```
